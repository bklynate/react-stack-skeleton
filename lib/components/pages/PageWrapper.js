'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = PageWrapper;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Header = require('../styles/Header');

var _Header2 = _interopRequireDefault(_Header);

var _Nav = require('../styles/Nav');

var _Nav2 = _interopRequireDefault(_Nav);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * The wrapper for ever page on the website.
 */
function PageWrapper(_ref) {
  var menuItems = _ref.menuItems;
  var children = _ref.children;

  var otherProps = _objectWithoutProperties(_ref, ['menuItems', 'children']);

  return _react2.default.createElement(
    'div',
    otherProps,
    _react2.default.createElement(_Header2.default, {
      title: 'React Stack Skeleton',
      menu: _react2.default.createElement(_Nav2.default, { items: menuItems })
    }),
    _react2.default.createElement(
      'div',
      null,
      children
    )
  );
}