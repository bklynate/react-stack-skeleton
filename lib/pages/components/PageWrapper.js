'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = PageWrapper;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _core = require('../../core');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * A wrapper for ever page on the site.
 */
function PageWrapper(_ref) {
  var children = _ref.children,
      otherProps = _objectWithoutProperties(_ref, ['children']);

  return _react2.default.createElement(
    'div',
    otherProps,
    _react2.default.createElement(_core.Header, null),
    children,
    _react2.default.createElement(_core.Footer, null)
  );
}