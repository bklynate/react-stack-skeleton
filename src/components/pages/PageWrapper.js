/* @flow */
import React from 'react';

import Header from '../styles/Header';
import Nav from '../styles/Nav';

import type { NavItem } from '../../stores/navigation';

type Props = {
  menuItems: Array<NavItem>,
  children?: any,
};

/**
 * The wrapper for ever page on the website.
 */
export default function PageWrapper({ menuItems, children, ...otherProps }: Props): React.Element<*> {
  return (
    <div {...otherProps}>
      <Header
        title="React Stack Skeleton"
        menu={<Nav items={menuItems} />}
      />
      <div>
        {children}
      </div>
    </div>
  );
}
