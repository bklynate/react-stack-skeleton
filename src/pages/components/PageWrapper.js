/* @flow */
import React from 'react';

type Props = {
  children?: any,
};

/**
 * A wrapper for ever page on the site.
 */
export default function PageWrapper({ children, ...otherProps }: Props): React.Element<*> {
  return (
    <div {...otherProps}>
      <header>NYI: Header</header>
      {children}
      <footer>NYI: Footer</footer>
    </div>
  );
}
