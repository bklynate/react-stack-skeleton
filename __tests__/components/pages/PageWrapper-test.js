import React from 'react';
import renderer from 'react-test-renderer';

import PageWrapper from '../../../src/components/pages/PageWrapper';

test('The PageWrapper component displays the given navigation items.', () => {
  const component = renderer.create(
    <PageWrapper
      menuItems={[
        { label: 'Home', to: '/' },
        { label: 'Internal Link', to: '/internal' },
        { label: 'External Link', href: 'https://google.com' },
      ]}
    />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('The PageWrapper component displays the given children.', () => {
  const component = renderer.create(
    <PageWrapper menuItems={[]}>
      <div>Test child</div>
    </PageWrapper>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
