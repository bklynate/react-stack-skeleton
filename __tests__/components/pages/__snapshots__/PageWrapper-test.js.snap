exports[`test The PageWrapper component displays the given children. 1`] = `
<div>
  <header
    style={
      Object {
        "backgroundColor": "black",
        "height": 150,
        "width": "100%",
      }
    }>
    <h1
      style={
        Object {
          "color": "white",
          "left": 10,
          "position": "absolute",
          "top": 10,
        }
      }>
      React Stack Skeleton
    </h1>
    <div
      style={
        Object {
          "position": "absolute",
          "right": 10,
          "top": 10,
        }
      }>
      <nav>
        <ul
          style={
            Object {
              "listStyleType": "none",
            }
          } />
      </nav>
    </div>
  </header>
  <div>
    <div>
      Test child
    </div>
  </div>
</div>
`;

exports[`test The PageWrapper component displays the given navigation items. 1`] = `
<div>
  <header
    style={
      Object {
        "backgroundColor": "black",
        "height": 150,
        "width": "100%",
      }
    }>
    <h1
      style={
        Object {
          "color": "white",
          "left": 10,
          "position": "absolute",
          "top": 10,
        }
      }>
      React Stack Skeleton
    </h1>
    <div
      style={
        Object {
          "position": "absolute",
          "right": 10,
          "top": 10,
        }
      }>
      <nav>
        <ul
          style={
            Object {
              "listStyleType": "none",
            }
          }>
          <li
            style={
              Object {
                "float": "left",
                "marginLeft": 25,
                "width": 100,
              }
            }>
            <a
              onClick={[Function]}
              style={Object {}}>
              Home
            </a>
          </li>
          <li
            style={
              Object {
                "float": "left",
                "marginLeft": 25,
                "width": 100,
              }
            }>
            <a
              onClick={[Function]}
              style={Object {}}>
              Internal Link
            </a>
          </li>
          <li
            style={
              Object {
                "float": "left",
                "marginLeft": 25,
                "width": 100,
              }
            }>
            <a
              href="https://google.com">
              External Link
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <div />
</div>
`;
